Antidote/AbstractCallViewController.m

```diff
@@ -0,0 +1,82 @@
+//
+//  AbstractCallViewController.m
+//  Antidote
+//
+//  Created by Chuong Vu on 7/2/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "AbstractCallViewController.h"
+
+#import "Masonry.h"
+#import "Helper.h"
+#import "OCTCall.h"
+#import "OCTChat.h"
+#import "OCTSubmanagerCalls.h"
+
+@interface AbstractCallViewController () <RBQFetchedResultsControllerDelegate>
+
+@property (strong, nonatomic) RBQFetchedResultsController *callController;
+
+@end
+
+@implementation AbstractCallViewController
+
+#pragma mark - Life cycle
+
+- (instancetype)initWithChat:(OCTChat *)chat submanagerCalls:(OCTSubmanagerCalls *)manager
+{
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"chat == %@", chat];
+    _callController = [Helper createFetchedResultsControllerForType:OCTFetchRequestTypeCall predicate:predicate delegate:self];
+    _callController.delegate = self;
+
+    _call = [[_callController fetchedObjects] firstObject];
+
+    if (! _call) {
+        return nil;
+    }
+
+    _manager = manager;
+
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    self.view.opaque = YES;
+
+    self.view.backgroundColor = [UIColor clearColor];
+
+    [self setupBlurredView];
+}
+
+- (void)setupBlurredView
+{
+    UIView *darkView = [[UIView alloc] initWithFrame:self.view.bounds];
+    darkView.backgroundColor = [UIColor blackColor];
+    darkView.alpha = 0.5;
+
+    [self.view addSubview:darkView];
+
+    [darkView mas_makeConstraints:^(MASConstraintMaker *make) {
```

Comment:
btw, you can use Masonry's methods without `mas_` prefix. Just `makeConstraints:`